numba/core/types/iterators.py

```diff
@@ -96,7 +97,9 @@ def __init__(self, array_type):
         self.array_type = array_type
         name = "iter(%s)" % (self.array_type,)
         nd = array_type.ndim
-        if nd == 0 or nd == 1:
+        if nd == 0:
+            raise TypingError("iteration over a 0-d array")
```

Comment:
Note that by accident, cpython supports `iter(arr_0d)`, but then raises an exception when you call `next(it)`.

Reported upstream at https://bugs.python.org/issue39610.

Reply:
Great, thanks for reporting upstream.